opentracing-web-servlet-filter/src/main/java/io/opentracing/contrib/web/servlet/filter/TracingFilter.java

```
@@ -0,0 +1,159 @@
+    }
+
+    @Override
+    public void init(FilterConfig filterConfig) throws ServletException {
+        this.filterConfig = filterConfig;
+
+        // init if the tracer is null
+        if (null == tracer) {
+            ServletContext servletContext = filterConfig.getServletContext();
+            this.tracer = (Tracer)servletContext.getAttribute(TRACER);
+            this.tracingDecision = (TracingDecision)servletContext.getAttribute(TRACING_DECISION);
+            if (null == this.tracingDecision) {
+                this.tracingDecision = TracingDecision.TRACE_ALL;
+            }
+            this.spanDecorators = (List<SpanDecorator>)servletContext.getAttribute(SPAN_DECORATORS);
+            if (null == this.spanDecorators) {
+                this.spanDecorators = Arrays.asList(SpanDecorator.STANDARD_TAGS);
+            }
+        }
+    }
+
+    @Override
+    public void doFilter(ServletRequest servletRequest, ServletResponse servletResponse, FilterChain chain)
+            throws IOException, ServletException {
+
+        HttpServletRequest httpRequest = (HttpServletRequest) servletRequest;
+        HttpServletResponse httpResponse = (HttpServletResponse) servletResponse;
+
+        /**
+         * Check if request has been already traced do not proceed.
+         */
+        if (null != servletRequest.getAttribute(CURRENT_SPAN)) {
+            chain.doFilter(servletRequest, servletResponse);
+        } else if (tracingDecision.isTraced(httpRequest, httpResponse)){
+            SpanContext extractedContext = tracer.extract(Format.Builtin.HTTP_HEADERS,
+                    new HttpServletHeadersExtractAdapter(httpRequest));
+
+            Span span = tracer.buildSpan(httpRequest.getMethod())
+                    .asChildOf(extractedContext)
+                    .withTag(Tags.SPAN_KIND.getKey(), Tags.SPAN_KIND_SERVER)
+                    .start();
+
+            for (SpanDecorator spanDecorator : spanDecorators) {
```

Yes, everything that can go wrong must be checked, specially on places that is the interface between your code and "external" code. Of course, on "internal APIs" it's OK to not do the null check and let NPE's happen, but it's usually bad practice to let your consumer face a NPE: it's better to throw a new exception, stating exactly what is wrong.